src/catalogue/app/src/main/java/catalogue/Application.java

```diff
@@ -32,6 +33,9 @@
 public class Application {
 
     public static void main(String[] args) {
-        Micronaut.run(Application.class, args);
+        Micronaut.build(args)
+                .mainClass(Application.class)
+                .defaultEnvironments(Environment.DEVELOPMENT)
```

Comment:
Why is this needed?

Reply:
Ok, I see now that you put the DB configuration in an environment-specific config file.

Defaulting the environment in `Application` is a bad practise. The environment should be defined when running the application. It could be part of the entrypoint of the `Dockerfile` (`java -Dmicronaut.environments=dev..`), but in this case it is much better to define the `MICRONAUT_ENVIRONMENTS` environment variable in the Kubernetes deployment manifest.

Reply:
Also, the `dev` name is misleading, since you are using that to deploy to "production". Perhaps `in-memory` or `h2` are better alternatives.

Reply:
@alvarosanchez @graemerocher - Maybe we should use `app` instead of `h2` or something else that is not specific just to `h2` database because I need to setup additional configuration just for that environment.

Currently the `app` subproject has
```
  metrics:
    export:
      prometheus:
        enabled: true
        descriptions: true
        step: PT1M
```
in the `application.yml` file. But that configuration should be in the `application-h2.yml` file since it is specific only for the `app` subproject (should not be used in `aws` and `oci` subprojects).

Also the following needs to be added to the `app` subproject configuration
```
endpoints:  
    prometheus:
        enabled: true
        sensitive: false
        details-visible: ANONYMOUS
```